rocky/rocky/scheduler.py

```diff
@@ -182,17 +185,25 @@ class ConflictError(SchedulerError):
     message = _("Task already queued.")
 
 
+class TaskNotFoundError(SchedulerError):
+    message = _("Task could not be found.")
+
+
 class SchedulerClient:
     def __init__(self, base_uri: str):
         self.session = requests.Session()
```

Comment:
Could we add a small middleware class that offers the following method to inject the error handling as we use it now?

```
class SchedulerSession(requests.Session):
  def request(self, *args, **kwargs):
    try:
      return super().request(*args, **kwargs)
    except (requests.ConnectionError, requests.RequestException, requests.TooManyRedirects, requests.ConnectTimeout, requests.ReadTimeout, reuqests.Timeout) as error:
      raise SchedulerError(error)
```

Reply:
I've found a midway solution.